lineman/app/components/thread_page/thread_page_controller.coffee

```diff
@@ -1,24 +1,48 @@
 angular.module('loomioApp').controller 'ThreadPageController', ($scope, $routeParams, $location, $rootScope, Records, MessageChannelService, CurrentUser, DiscussionFormService, ScrollService) ->
   $rootScope.$broadcast('currentComponent', { page: 'threadPage'})
 
-  $scope.$on 'threadPageEventsLoaded',    (event, sequenceId) =>
-    ScrollService.scrollTo("#sequence-#{sequenceId}") if @focusMode == 'activity'
-  $scope.$on 'threadPageProposalsLoaded', (event, proposalId) =>
-    ScrollService.scrollTo("#proposal-#{proposalId}") if @focusMode == 'proposal'
-
-  Records.discussions.findOrFetchByKey($routeParams.key).then (discussion) =>
-    @discussion = discussion
-    $rootScope.$broadcast('setTitle', @discussion.title)
-    @group = @discussion.group()
-    $rootScope.$broadcast('viewingThread', @discussion)
-    MessageChannelService.subscribeTo("/discussion-#{@discussion.key}", @onMessageReceived)
-    @setFocusMode()
-    ScrollService.scrollTo('.thread-context') if @focusMode == 'context'
-  , (error) ->
+  @performScroll = ->
+    elementToFocus = @elementToFocus()
+    if elementToFocus && !@scrolledAlready
+      ScrollService.scrollTo elementToFocus
+      @scrolledAlready = true
+
+  @elementToFocus = ->
+    if $location.hash().match(/^proposal-\d+$/) and Records.proposals.find(@focusedProposalId)
+      "#proposal-#{@focusedProposalId}"
+    else if @discussion.lastSequenceId == 0 or @sequenceIdToFocus == -1
+      ".thread-context"
+    else if Records.events.findByDiscussionAndSequenceId(@discussion, @sequenceIdToFocus)
+      if @discussion.isUnread()
+        ".activity-card__new-activity"
+      else
+        "#sequence-#{@sequenceIdToFocus}"
+    else
+      false # our record isn't there yet
+
+  @init = (discussion) =>
+    if discussion and !@discussion?
+      @discussion = discussion
+      @group = @discussion.group()
+      @sequenceIdToFocus = @discussion.reader().lastReadSequenceId # or location hash when we put it back in.
```

Comment:
Yea I totally agree... It's a nice way to sidestep the sequence Id problems too..because the thread ng-repeat can be responsible for placing the element we're going to scroll+focus on, and we just reference that rahter than an ID.

The only thing I'd like to add is that we focus on the new activity indicator.. or if there is no new activity then focus on the last item. (i suggest we add a 'last item' class). I reckon it will be important for a11y to focus on the new activity indicator.. that way the screenreader can say stuff like "5 new comments since you last read this discussion" and they can continue on happy and knowing what's going on.